Add the 'constexpr' keyword to C++ named patterns

M4444 opened the pull request:
## Description
A header file in my repo was misclassified as C instead of C++:
https://github.com/M4444/TMatrix/search?l=c
https://github.com/M4444/TMatrix/blob/43cd9ce1830bb7e78b7746972296cd3fc368b1e9/include/tmatrix.h

The file contains the `constexpr` keyword which only exists in C++.
This change is the simplest one that fixes the problem.

## Checklist:
- [x] **I am fixing a misclassified language**
  - [x] I have included a change to the heuristics to distinguish my language from others using the same extension.

pchaigno commented on lib/linguist/heuristics.yml:
```diff
@@ -460,6 +460,7 @@ named_patterns:
   - '^[ \t]*catch\s*\('
   - '^[ \t]*(class|(using[ \t]+)?namespace)\s+\w+'
   - '^[ \t]*(private|public|protected):$'
+  - '^[ \t]*constexpr'
```
Couldn't we merge this pattern with the `try` one above?

M4444 commented:
Addressed the comments. The file in the mentioned repo is under GPL-2.0-only so I created a header file with a similar structure for the sample.
I've pushed a new commit for this because I'm now sure what effect squashing them would have on the pull request but you can squash them before they get merged.
